lib/modules/manager/npm/extract/common/dependency.ts

```diff
@@ -180,6 +180,16 @@ export function extractDependency(
     dep.datasource = GithubTagsDatasource.id;
     dep.versioning = npmVersioningId;
     dep.packageName = githubOwnerRepo;
+  } else if (
+    depRefPart.startsWith('semver:') &&
+    isVersion(depRefPart.slice('semver:'.length))
```

Comment:
use substring (perf)

Reply:
I was not aware of a performance difference but I updated the code to only extract the version once.